library/ssl_tls.c

```diff
@@ -275,6 +275,17 @@ int mbedtls_ssl_session_copy(mbedtls_ssl_session *dst,
 
 #endif /* MBEDTLS_X509_CRT_PARSE_C */
 
+#if defined(MBEDTLS_SSL_SRV_C) && defined(MBEDTLS_SSL_ALPN) && \
+    defined(MBEDTLS_SSL_EARLY_DATA)
+    {
+        int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+        ret = mbedtls_ssl_session_set_ticket_alpn(dst, src->ticket_alpn);
```

Comment:
Missing `dst->ticket_alpn = NULL` at the beginning of the function (see what is done for `dst->hostname`) leading to a dangling pointer it seems.